lambdas/authorisation/src/test/java/uk/gov/di/data/lep/AuthorisationTest.java

```diff
@@ -4,32 +4,70 @@
+import com.auth0.jwt.interfaces.DecodedJWT;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import uk.gov.di.data.lep.library.config.Config;
+import uk.gov.di.data.lep.services.JwtService;
 
 import java.util.Map;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.clearInvocations;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
 public class AuthorisationTest {
-    private static final Authorisation underTest = new Authorisation();
+    private static final Config config = mock(Config.class);
+    private static final JwtService jwtService = mock(JwtService.class);
+    private static final Authorisation underTest = new Authorisation(config, jwtService);
     private static final Context context = mock(Context.class);
+    private static final DecodedJWT decodedJwt = mock(DecodedJWT.class);
+
+    @BeforeEach
+    void refreshSetup() {
+        clearInvocations(config);
+        clearInvocations(context);
+        clearInvocations(decodedJwt);
+        clearInvocations(jwtService);
+    }
 
     @Test
     void authorisationReturnsAnAuthorisorResponseWithCorrectActionAndEffect() {
+        when(config.getAwsRegion()).thenReturn("eu-west-2");
+        when(config.getUserPoolId()).thenReturn("userPoolId");
+        when(context.getIdentity()).thenReturn(null);
+        when(decodedJwt.getSubject()).thenReturn("testSubject");
+        when(jwtService.decode(anyString())).thenReturn(decodedJwt);
+
         var event = new APIGatewayProxyRequestEvent()
             .withHeaders(
                 Map.of(
                     "Authorization",
-                    "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c"))
+                    "Bearer accessToken"))
             .withRequestContext(new ProxyRequestContext().withIdentity(new RequestIdentity()));
 
-        when(context.getIdentity()).thenReturn(null);
-
         var result = underTest.handleRequest(event, context);
 
         assertEquals("execute-api:Invoke", result.policyDocument.Statement.get(0).Action);
         assertEquals("Allow", result.policyDocument.Statement.get(0).Effect);
     }
+
+    @Test
+    void authorisationThrowsAnError() {
+        when(jwtService.decode(anyString())).thenThrow(new JWTVerificationException("Exception message"));
+
+        var event = new APIGatewayProxyRequestEvent()
+            .withHeaders(
+                Map.of(
+                    "Authorization",
+                    "Bearer accessToken"))
+            .withRequestContext(new ProxyRequestContext().withIdentity(new RequestIdentity()));
+
+        assertThrows(RuntimeException.class, () -> underTest.handleRequest(event, context));
```

Comment:
Could we assert some details of the throw please, you should be able to verify the inner exception of it